source-rentcast contribution from ombhardwajj

ombhardwajj opened the pull request:
This closes #46884
## What

This PR adds a new connector for source RentCast (source-rentcast).

Contributor-provided connector description:

RentCast is the leading rental property analytics, estimation and reporting software .
This connector enables you to extract data from endpoints like Value Estimate , Rent Estimate , Property Records , Sale Listings and Rental Listings
Docs : https://developers.rentcast.io/reference/introduction
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `api_key` | `string` | API Key.  |  |
| `address` | `string` | Address. The full address of the property, in the format of Street, City, State, Zip. Used to retrieve data for a specific property, or together with the radius parameter to search for listings in a specific area |  |
| `city` | `string` | City. The name of the city, used to search for listings in a specific city. This parameter is case-sensitive |  |
| `state` | `string` | State. The 2-character state abbreviation, used to search for listings in a specific state. This parameter is case-sensitive |  |
| `zipcode` | `string` | Zip Code. The 5-digit zip code, used to search for listings in a specific zip code |  |
| `latitude` | `string` | Latitude. The latitude of the search area. Use the latitude/longitude and radius parameters to search for listings in a specific area |  |
| `longitude` | `string` | Longitude. The longitude of the search area. Use the latitude/longitude and radius parameters to search for listings in a specific area |  |
| `radius` | `string` | Radius. The radius of the search area in miles, with a maximum of 100. Use in combination with the latitude/longitude or address parameters to search for listings in a specific area |  |
| `property_type` | `string` | Property Type. The type of the property, used to search for listings matching this criteria : Single Family , Condo , Townhouse , Manufactured ,  Multi-Family , Apartment , Land , |  |
| `bedrooms` | `number` | Bedrooms. The number of bedrooms, used to search for listings matching this criteria. Use 0 to indicate a studio layout |  |
| `bath_rooms` | `integer` | Bath Rooms. The number of bathrooms, used to search for listings matching this criteria. Supports fractions to indicate partial bathrooms |  |
| `status` | `string` | Status. The current listing status, used to search for listings matching this criteria : Active or Inactive |  |
| `days_old` | `string` | Days Old. The maximum number of days since a property was listed on the market, with a minimum of 1 or The maximum number of days since a property was last sold, with a minimum of 1. Used to search for properties that were sold within the specified date range |  |
| `data_type_` | `string` | Data Type . The type of aggregate market data to return. Defaults to &quot;All&quot; if not provided : All , Sale , Rental |  |
| `history_range` | `string` | History Range. The time range for historical record entries, in months. Defaults to 12 if not provided |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| Property Records | id | DefaultPaginator | ✅ |  ❌  |
| Sale Listings | id | DefaultPaginator | ✅ |  ❌  |
| Rental Listings | id | No pagination | ✅ |  ❌  |
| Statistics |  | No pagination | ✅ |  ❌  |
| Value Estimate |  | No pagination | ✅ |  ❌  |
| Rent Estimate |  | No pagination | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

<!-- DO NOT REMOVE: connector-builder-new-connector-contribution -->

marcosmarxm reviewed (changes requested):
Stream `property_records` please add all optional parameters to this stream as the sale one.

Stream `sales_listings` is not working.

Please remove:
- `property_record`
- `sale_listing`
- `rental_listing`
They are basically the same as the other ones but retrieve a single record.

Some endpoints required parameters, example `value_estimate` must have an address. Please add an error handle when the user don't provide the value. The same to all endpoints :)

ombhardwajj commented:
@marcosmarxm 
1) I've added all the parameters to the Stream `property_records`
2) I again tested the stream `sales_listings` its working fine for me...can you test it from my account? Or let me know the query paramters you attach so that I can test it out with your request.
3) Removed these streams `property_record` `sale_listing` `rental_listing`
4) Added the error handling for the two required streams

marcosmarxm commented:
Hello 👋 Due to the numerous contributions happening, I decided to create this [view](https://github.com/orgs/airbytehq/projects/104/views/3) to help you know when I plan to review your contribution. If your connector has some limited-time credential feel free to reach me out.
